crates/syntax/src/ast/make.rs

```diff
@@ -15,12 +15,22 @@ use stdx::format_to;
 use crate::{ast, AstNode, SourceFile, SyntaxKind, SyntaxNode, SyntaxToken};
 
 pub fn name(text: &str) -> ast::Name {
-    ast_from_text(&format!("mod {};", text))
+    ast_from_text(&format!("mod {}{};", raw_ident_esc(text), text))
 }
 
 pub fn name_ref(text: &str) -> ast::NameRef {
-    ast_from_text(&format!("fn f() {{ {}; }}", text))
+    ast_from_text(&format!("fn f() {{ {}{}; }}", raw_ident_esc(text), text))
 }
+
+fn raw_ident_esc(ident: &str) -> &'static str {
+    let is_keyword = parser::SyntaxKind::from_keyword(ident).is_some();
+    if is_keyword && !matches!(ident, "self" | "crate" | "super" | "Self") {
```

Comment:
Seems that `Self` is an odd one out here? (or `from_keyword` is really weird?)

Reply:
Yeah, these are taken from https://doc.rust-lang.org/reference/identifiers.html#identifiers